www/src/app/views/InboxContainer.js

```diff
@@ -30,114 +20,74 @@ export default class InboxContainer extends Component {
 
     this.handleResolve = this.handleResolve.bind(this);
     this.handleArchive = this.handleArchive.bind(this);
+
+    this.state = {
+      editText: '',
+      updates: []
+    };
   }
 
-  componentDidMount() {
-    const { author } = this.state;
-    const currentReportsInfo = `${API_URL}/reports/current`;
-    const updatesByAuthor = `${API_URL}/updates?author=${author}`;
-
-    get(currentReportsInfo).then(
-      ([prevReport, nextReport]) => Promise.all([
-        prevReport, nextReport,
-        get(`${updatesByAuthor}&resolved=0&status=inbox&status=event&status=todo&status=done`),
-        get(`${updatesByAuthor}&resolved=0&status=goal&before=${nextReport.slug}`),
-        get(`${updatesByAuthor}&report=${nextReport.slug}&status=goal&status=struggle&status=achievement`),
-      ])
-    ).then(
-      ([prevReport, nextReport, current, prev, next]) => this.setState({
-        prevReportDate: new Date(prevReport.reportDate),
-        nextReportDate: new Date(nextReport.reportDate),
-        nextReportSlug: nextReport.slug,
-        updates: [...current, ...prev, ...next].map(makeUpdate)
-      })
-    ).catch(console.error);
+  componentWillMount() {
+    const { params, setAuthor } = this.props;
+    const { fetchCurrentReports, fetchUpdatesByAuthor } = this.props;
+
+    setAuthor(params.author);
+    fetchCurrentReports().then(reports => {
+      fetchUpdatesByAuthor(params.author, reports.nextReportSlug)
+    });
```

Comment:
This is again subjective, but for one-liners like this I prefer the syntax without braces:

    fetchCurrentReports().then(
        reports => fetchUpdatesByAuthor(params.author, reports.nextReportSlug)
    );